coral-hive/src/main/java/com/linkedin/coral/transformers/OneBasedArrayIndexTransformer.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2022-2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.transformers;
+
+import java.math.BigDecimal;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNumericLiteral;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.type.ArraySqlType;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.apache.calcite.sql.validate.SqlValidator;
+
+import com.linkedin.coral.common.transformers.OperatorTransformer;
+
+
+/**
+ * Transformer to convert SqlCall from array[i] to array[i+1] to ensure array indexes start at 1.
+ */
+public class OneBasedArrayIndexTransformer extends OperatorTransformer {
+
+  private static final String ITEM_OPERATOR = "ITEM";
+
+  public OneBasedArrayIndexTransformer(SqlValidator sqlValidator) {
```

Comment:
Why does the caller need to provide the SqlValidator? Is there a way to set it automatically?

Reply:
In order to derive the datatype, we need to validate the SqlNode first, and SqlValidator contains necessary information (like table schema in HMS, and legal operators in `DaliOperatorTable`) for type derivation.